flink-runtime/src/test/java/org/apache/flink/runtime/testutils/InMemoryReporterRule.java

```diff
@@ -41,26 +58,41 @@
  */
 @Experimental
 public class InMemoryReporterRule implements ExternalResource {
-    private final InMemoryReporter inMemoryReporter = InMemoryReporter.getInstance();
+    private final InMemoryReporter inMemoryReporter;
 
-    public static InMemoryReporterRule create() {
-        return new InMemoryReporterRule();
+    public static InMemoryReporterRule fromMiniCluster(MiniClusterResource miniClusterResource) {
```

Comment:
Can we invert the dependency? Differently said: Can we allow to configure a `MetricReporter` for the `MiniCluster`? If yes, then we can have our `InMemoryReporterRule` (actually so far all `ExternalResources` have the `Resource` suffix and `Rule`), which creates our reporter and then we start the `MiniCluster` with this reporter. That way we can also have a special `MetricReporter` that retains the metrics.